### PR TITLE
[GNA] Move weights transposition from NCHW to NHWC to GNA pass after FQ passes

### PR DESCRIPTION
**Pull request by elilobanova**
### Details:
 - Currently weights transposition is performed before GNA passes, if the model contains fake quantizes for weighted layers their weights will be kept in additional constants but not in CNN layer data structure. This transformation should be done after the pass which fuses quantized weights with their layer structure.

### Tickets:
- 49993

### CODE REVIEW

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.hpp [hunk 1]
```diff
@@ -209,6 +209,11 @@ DECL_PASS(FuseFQIntoWeights);
 */
 DECL_PASS(MoveFakeQuantizeLayerIntoQuantParams);
 
+/**
+* @brief convert FullyConnected, ScaleShift and Eltwise layers weights order from NCHW to NHWC.
+* Information for transposition is found from convolution/pooling input or output dimensions.
+*/
+DECL_PASS(TransposeWeightsFromNCHWToNHWC);
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.hpp:216]: What about transposing weights (filters) of Convolution layer itself is it covered here or not?
And why/ why not ?

**Review by kbruniec — COMMENTED** [review]
As far as I understand it is OK, but more clarifications (comments) would be more than welcome :).

File: inference-engine/src/gna_plugin/gna_tensor_tools.hpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * @brief transposes a tensor from NCHW to NHWC order
+ * @param precision data precision
+ * @param rows number of rows in tensor
+ * @param columns number of columns in tensor
+ * @param buffer pointer to a tensor buffer
+ * @param transpose_rows flag indicated if tensor rows must be transposed
+ * @param transpositionInfo information about transposition
+ */
+inline void TransposeTensorFromNCHWToNHWC(size_t precision, size_t rows, size_t columns, uint8_t* buffer, bool transpose_rows,
+                                          const std::vector<TranspositionInfo> &transpositionInfo) {
```
[the remark this answers is not in the document]
Comment (elilobanova) [reply inference-engine/src/gna_plugin/gna_tensor_tools.hpp]: I've updated the function description, please let me know if something is still not clear.

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.hpp (hunk 1 above)
Comment (kbruniec) [reply inference-engine/src/gna_plugin/optimizer/gna_pass_manager.hpp:216]: Thanks, consider to add it as comment.

File: inference-engine/src/gna_plugin/gna_tensor_tools.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (kbruniec) [reply inference-engine/src/gna_plugin/gna_tensor_tools.hpp]: Is is OK now. Thanks

**Review by kbruniec — APPROVED** [review]

**Pull request merged**